addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/api/manager/MiHomeApiManagerImpl.java

```diff
@@ -0,0 +1,204 @@
+
+        return execute(CONTROLLER_DEVICES, ACTION_UPDATE_FIRMWARE, parameters);
+    }
+
+    @Override
+    public JsonObject listSubdevices() {
+        return execute(CONTROLLER_SUBDEVICES, ACTION_LIST);
+    }
+
+    @Override
+    public JsonObject registerSubdevice(int gatewayID, String deviceType) {
+        Map<String, Object> parameters = new HashMap<String, Object>();
+        parameters.put(DeviceConstants.SUBDEVICE_PARENT_ID_KEY, gatewayID);
+        parameters.put(DeviceConstants.DEVICE_TYPE_KEY, deviceType);
+        return execute(CONTROLLER_SUBDEVICES, ACTION_CREATE, parameters);
+
+    }
+
+    @Override
+    public JsonObject showSubdeviceInfo(int id) {
+        Map<String, Object> parameters = new HashMap<String, Object>();
+        parameters.put(DeviceConstants.DEVICE_ID_KEY, id);
+        // With this parameter we exclude the historical usage data ("1" - to include, "0" - to exclude)
+        // The historical data is from no importance to the user,
+        // it consists of the number of events for each hour from the last 24 hours.
+        parameters.put(DeviceConstants.SUBDEVICE_INCLUDE_USAGE_DATA, 0);
+        return execute(CONTROLLER_SUBDEVICES, ACTION_SHOW, parameters);
+    }
+
+    @Override
+    public JsonObject updateSubdevice(int id, String label) {
+        Map<String, Object> parameters = new HashMap<String, Object>();
+        parameters.put(DeviceConstants.DEVICE_ID_KEY, id);
+        parameters.put(DeviceConstants.DEVICE_LABEL_KEY, label);
+        return execute(CONTROLLER_SUBDEVICES, ACTION_UPDATE, parameters);
+    }
+
+    @Override
+    public JsonObject unregisterSubdevice(int id) {
+        Map<String, Object> parameters = new HashMap<String, Object>();
+        parameters.put(DeviceConstants.DEVICE_ID_KEY, id);
+        return execute(CONTROLLER_SUBDEVICES, ACTION_DELETE, parameters);
+    }
+
+    private JsonObject execute(String controller, String action) {
+        return execute(controller, action, new HashMap<String, Object>());
+    }
+
+    private JsonObject execute(String controller, String action, Map<String, Object> content) {
+
+        Properties httpHeaders = new Properties();
+
+        // Set basic authentication header
+        String authData = configuration.getUserName() + ":" + configuration.getPassword();
+        String basicAuthentication = "Basic " + Base64.getEncoder().encodeToString(authData.getBytes());
+        basicAuthentication = basicAuthentication.trim();
+        httpHeaders.setProperty("Authorization", basicAuthentication);
+
+        // Convert request the content to JSON
+        Gson gson = new Gson();
+        String jsonContent = gson.toJson(content);
+        InputStream requestContent = IOUtils.toInputStream(jsonContent);
+        ContentResponse contentResponse = null;
+        String responseBody = null;
+
+        try {
+            contentResponse = restClient.sendRequest(controller + "/" + action, RestClient.DEFAULT_HTTP_METHOD,
```

Comment:
RestClient.DEFAULT_HTTP_METHOD => most people with some rest/http experience method would expect this to be GET, so just inline so that it's clear that you always sending post requests.